Change

akshat01112001 opened the pull request.

akshat01112001 committed:
Done

akshat01112001 committed:
Create .gitignore

akshat01112001 committed:
Update package.json

akshat01112001 committed:
Update Page.ts

akshat01112001 committed:
Update FileChooser.ts

akshat01112001 committed:
Update ExecutionContext.ts

akshat01112001 committed:
Update EventEmitter.ts

akshat01112001 committed:
Update ConsoleMessage.ts

akshat01112001 committed:
Update Connection.ts

akshat01112001 committed:
Update Browser.ts

akshat01112001 committed:
Merge branch 'main' into Change

akshat01112001 committed:
Merge branch 'main' into Change

akshat01112001 committed:
Update Browser.ts

akshat01112001 committed:
Merge branch 'Change' of https://github.com/akshat01112001/puppeteer into Change

akshat01112001 committed:
Merge branch 'main' into Change

jackfranklin commented on .husky/.gitignore:
```diff
@@ -0,0 +1 @@
+_
```
Do you know what caused this change?

jackfranklin commented on package.json:
```diff
@@ -109,6 +110,7 @@
     "standard-version": "^9.0.0",
     "text-diff": "^1.0.1",
     "ts-node": "^9.0.0",
+    "typedoc": "^0.20.32",
```
typedoc is different to TSDoc (confusingly!) so I don't think we want this. Why did you add it?

jackfranklin commented on src/common/Browser.ts:
```diff
@@ -412,7 +412,7 @@ export class Browser extends EventEmitter {
       browserContextId: contextId || undefined,
     });
-    const target = await this._targets.get(targetId);
+    const target = this._targets.get(targetId);
```
Nice spot!

jackfranklin commented on src/common/Connection.ts:
```diff
@@ -67,8 +67,8 @@ export class Connection extends EventEmitter {
-   * @param {string} sessionId
-   * @returns {?CDPSession}
+   * @param sessionId - Gets Session ID
+   * @returns - Gets CDPSession
```
You don't need a dash after `@returns` - see https://tsdoc.org/pages/tags/returns/

So this should be

```
@returns the current CDP session if it exists

jackfranklin commented on src/common/Connection.ts:
```diff
@@ -167,8 +167,8 @@ export class Connection extends EventEmitter {
-   * @param {Protocol.Target.TargetInfo} targetInfo
-   * @returns {!Promise<!CDPSession>}
+   * @param targetInfo - Gets targetInfo
+   * @returns sessionID
```
try and write these like a sentence, so this might be:

```
@returns the CDP session that is created
```

jackfranklin commented on src/common/Connection.ts:
```diff
@@ -167,8 +167,8 @@ export class Connection extends EventEmitter {
   /**
-   * @param {Protocol.Target.TargetInfo} targetInfo
-   * @returns {!Promise<!CDPSession>}
+   * @param targetInfo - Gets targetInfo
```
targetInfo here is a param the user provides, so "Gets targetInfo" doesn't quite fit. If you're not sure what to document for this param, just leave it as:

```
@param targetInfo
```

jackfranklin commented on src/common/EventEmitter.ts:
```diff
@@ -33,7 +33,7 @@ export interface CommonEventEmitter {
  * and unbind to event listeners.
  *
- * @public
+ * @internal
```
why did you change this? We do want to document this class so I don't believe this should be internal.

jackfranklin commented on src/common/ExecutionContext.ts:
```diff
@@ -33,7 +33,7 @@ const SOURCE_URL_REGEX = /^[\040\t]*\/\/[@#] sourceURL=\s*(\S*?)\s*$/m;
  *   frame } has "default" execution context that is always created after frame is
  *   attached to DOM. This context is returned by the
- *   {@link frame.executionContext()} method.
+ *   {@link frame.executionContext()| method} method.
```
this will case the rendered text to read `method method` - the bit after the `|` is the text used. Why did you make this change?

jackfranklin replied:
@akshat01112001 please could you undo this change?

akshat01112001 replied:
I have undone this change

jackfranklin commented on src/common/FileChooser.ts:
```diff
@@ -33,7 +33,8 @@ import { assert } from './assert.js';
  * All file choosers must be accepted or canceled. Not doing so will prevent
- * subsequent file choosers from appearing.
+ * subsequent file choosers from app/earing.
+ * @internal
```
I think this should actually be `@public` as we want it documented to the user.

jackfranklin commented on src/common/Page.ts:
```diff
@@ -283,7 +283,7 @@ export const enum PageEmittedEvents {
    *
    * @remarks
-   * The object is readonly. See {@Page.setRequestInterception} for intercepting
+   * The object is readonly. See /{@Page setRequestInterception \} for intercepting
```
why did you add the slashes here?

jackfranklin reviewed (commented):
Thanks for your PR! I've left some comments with some questions, if you wouldn't mind taking another look 👍

akshat01112001 committed:
Update package.json

akshat01112001 committed:
Delete .gitignore

akshat01112001 committed:
Update Connection.ts

akshat01112001 committed:
Merge branch 'Change' of https://github.com/akshat01112001/puppeteer into Change

akshat01112001 committed:
Update FileChooser.ts

akshat01112001 committed:
Update EventEmitter.ts

akshat01112001 committed:
Update Connection.ts

akshat01112001 committed:
Update FileChooser.ts

akshat01112001 committed:
Update Page.ts

akshat01112001 committed:
Merge branch 'main' into Change

akshat01112001 commented:
I have updated the changes. Have a look.

Cheers,
Akshat

jackfranklin commented on src/common/Page.ts:
```diff
@@ -283,7 +283,7 @@ export const enum PageEmittedEvents {
    *
    * @remarks
-   * The object is readonly. See {@Page.setRequestInterception} for intercepting
+   * The object is readonly. See {@Page setRequestInterception} for intercepting
```
We still need the dot here - this should be `@Page.setRequestInterception` so TSDoc links to the right method

akshat01112001 committed:
Update Page.ts

akshat01112001 committed:
Update ExecutionContext.ts

galr52 committed:
captureBeyondViewport as screenshot optinal param (#7063)

* captureBeyondViewport as screenshot optinal param

* lint fixes

* fix lint

* apply Emulation.setDeviceMetricsOverride in
case captureBeyondViewport is false

kblok committed:
test: fix Browser.pages should return all of the pages (#7064)

karlhorky committed:
feat(types): improve type of predicate function (#6997)

From the DefinitelyTyped @types/puppeteer package:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c43191a8f7a7d2a47bbff0bc3a7d95ecc64d2269/types/puppeteer/index.d.ts#L1883-L1885

akshat01112001 committed:
Merge branch 'main' into Change

jackfranklin committed:
Merge branch 'main' into Change

akshat01112001 committed:
Merge branch 'main' into Change

akshat01112001 committed:
Update Browser.ts

akshat01112001 committed:
Merge branch 'Change' of https://github.com/akshat01112001/puppeteer into Change
